### DISCUSSION TITLE
properly mount cache for npm

### DISCUSSION THREAD

**Discussion by flixman** [containers/podman#22979] [Q&A]
I am coming from docker, and moving to podman. In my dockerfile, I have the following statement ```RUN --mount=type=cache,target=/root/.npm npm install```. I am running this on a podman-enabled Gitea runner, thus running on a podman-in-podman scenario. For what I have seen, in case TMPDIR is not defined (the case), I should get the /root/.npm folder in the building container mapped to ```/var/tmp/buildah-cache```, which I am mapping in the outer container to a folder in the host... but I see that empty. While debugging this, I have observed that there is a chroot involved so... I do not see how to proceed to have this cache persisted.

Can anybody provide some help?

**Comment by nalind** [comment]
IIRC, since around 1.29 it's defaulted to `/var/tmp/buildah-cache-$(id -u)`.  Can you try mounting the host directory there instead?

**Reply by flixman** [reply]
I have just discovered that :-D now I am facing another problem, that is that the folder is owned by root (while the user running the container is 1000). I have created an issue in gitea, because I think I am doing something wrong there. Thank you anyway!
